examples/data-solutions/cloudsql-multiregion/cloudsql.tf

```diff
@@ -16,8 +16,8 @@ module "db" {
   source              = "../../../modules/cloudsql-instance"
   project_id          = module.project.project_id
   availability_type   = var.sql_configuration.availability_type
-  encryption_key_name = var.cmek_encryption ? module.kms[var.regions.primary].keys.key.id : null
-  network             = module.vpc.self_link
+  encryption_key_name = var.service_encryption_keys != null ? var.service_encryption_keys[var.regions.primary] : null
```

Comment:
This might fail if Terraform still evaluates both sides regardless of the condition. Can you wrap the map access in `try`?